File: CapStone_Java/src/test/java/pageObjects/LoginPage.java
package pageObjects;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class LoginPage {
	WebDriver driver;
	public LoginPage(WebDriver driver) {
		this.driver = driver;
	}
	By userName = By.xpath("//input[@id='user-name']");
	By password = By.xpath("//input[@id='password']");
	By errorMsg = By.cssSelector("h3[data-test='error']");
	By submit = By.xpath("//input[@id='login-button']");
	
	public void setUserNamePassword(String username, String pass) {
		driver.findElement(userName).clear();
		driver.findElement(userName).sendKeys(username);
		driver.findElement(password).clear();
		driver.findElement(password).sendKeys(pass);
	}
	public String errorMsg() {
		new WebDriverWait(driver, Duration.ofSeconds(10)).
		until(ExpectedConditions.visibilityOfAllElementsLocatedBy(errorMsg));
		return driver.findElement(errorMsg).getText();
	}
	public void clickSubmit() {
		driver.findElement(submit).click();
	}
}

File: pythonProject/PageObjects/LoginPage.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


class LoginPage:

    def __init__(self, driver):
        self.driver = driver

    userName = (By.XPATH, "//input[@id='user-name']")
    password = (By.XPATH, "//input[@id='password']")
    submit = (By.XPATH, "//input[@id='login-button']")
    text = (By.CSS_SELECTOR, "h3[data-test='error']")

    def userNames(self):
        return self.driver.find_element(*LoginPage.userName)
    def passwords(self):
        return self.driver.find_element(*LoginPage.password)
    def submits(self):
        return self.driver.find_element(*LoginPage.submit)

    def texterror(self):
        element = WebDriverWait(self.driver, 10).until(EC.presence_of_element_located((By.CSS_SELECTOR,"h3[data-test='error']")))
        return element